src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -81,7 +135,106 @@ public T DeserializePayload<T>(Message message)
     {
         var versionedMessage = message as VersionedMessage;
         var payloadSerializer = GetPayloadSerializer(versionedMessage?.Version);
-        return Deserialize<T>(payloadSerializer, message.Payload);
+
+        if (!DisableFastJson)
```

Comment:
[cosmetic] Same suggestion about inverting the if and remove the else to reduce nesting.

Reply:
Inverted